packages/bsky/src/did-cache.ts

```diff
@@ -0,0 +1,71 @@
+import PQueue from 'p-queue'
+import { CacheResult, DidCache, DidDocument } from '@atproto/did-resolver'
+import Database from './db'
+
+export class DidSqlCache extends DidCache {
+  public pQueue: PQueue | null //null during teardown
+
+  constructor(public db: Database, public ttl: number) {
+    super()
+    this.pQueue = new PQueue()
+  }
+
+  async cacheDid(did: string, doc: DidDocument): Promise<void> {
+    await this.db.db
+      .insertInto('did_cache')
+      .values({ did, doc, updatedAt: Date.now() })
+      .onConflict((oc) =>
+        oc.column('did').doUpdateSet({ doc, updatedAt: Date.now() }),
+      )
+      .executeTakeFirst()
+  }
+
+  async refreshCache(
+    did: string,
+    getDoc: () => Promise<DidDocument | null>,
+  ): Promise<void> {
+    this.pQueue?.add(async () => {
+      const doc = await getDoc()
+      if (doc) {
+        await this.cacheDid(did, doc)
+      }
+    })
+  }
+
+  async checkCache(did: string): Promise<CacheResult | null> {
+    const res = await this.db.db
+      .selectFrom('did_cache')
+      .where('did', '=', did)
+      .selectAll()
+      .executeTakeFirst()
+    if (!res) return null
+    const now = Date.now()
+    const updatedAt = new Date(res.updatedAt).getTime()
```

Comment:
`Date` may not have to come into play here since `res.updatedAt` is a unix timestamp itself.

Reply:
oh yeah i had that as Date originally 🙃